fast_node.go

```diff
@@ -0,0 +1,41 @@
+package iavl
+
+import (
+	"github.com/pkg/errors"
+)
+
+// NOTE: This file favors int64 as opposed to int for size/counts.
+// The Tree on the other hand favors int.  This is intentional.
+
+type FastNode struct {
+	key                  []byte
+	versionLastUpdatedAt int64
+	value                []byte
+	leafHash             []byte // TODO: Look into if this would help with proof stuff.
+}
+
+// NewFastNode returns a new fast node from a value and version.
+func NewFastNode(value []byte, version int64) *FastNode {
```

Comment:
```suggestion
func NewFastNode(key []byte, value []byte, version int64) *FastNode {
```

I think it has to have key & value

Reply:
Yeah I noticed that the original node struct was initialized and then had it's hash set so followed that [pattern](https://github.com/cosmos/iavl/blob/577c9a8615e9edd17f2bba1c9c2b0547f8e96631/node.go#L44) with our fast node but looking into the call-sites for the original node struct constructor, I don't see any reason why that would be necessary.

Reply:
You're right, I overlooked this after adding the key field to the struct.